Add search functionality to query rule detail panel

efegurkan opened the pull request:
## Summary

Adds search element to the query rules detail page.

Summarize your PR. If it involves visual changes include a screenshot or gif.
<img width="2539" height="640" alt="Screenshot 2025-09-07 at 14 22 29" src="https://github.com/user-attachments/assets/25bd158b-8b49-4ceb-9467-ab100665eee1" />
<img width="2552" height="843" alt="Screenshot 2025-09-07 at 14 22 42" src="https://github.com/user-attachments/assets/868f5061-0803-409e-9bcf-12ec355d6588" />


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.






<!--ONMERGE {"backportTargets":["8.19","9.1"]} ONMERGE-->

efegurkan committed:
Add search functionality to query rule detail panel

efegurkan committed:
Merge branch 'main' of github.com:elastic/kibana into searchable-detail-panel

efegurkan committed:
Add result component

efegurkan committed:
Merge branch 'main' of github.com:elastic/kibana into searchable-detail-panel

efegurkan committed:
Fix small visual issues.

wildemat commented on x-pack/solutions/search/plugins/search_query_rules/public/components/query_ruleset_detail/query_rule_draggable_list/query_rule_search_results.tsx:
```diff
@@ -0,0 +1,262 @@
+                    <>
+                      {queryRule.criteria.slice(0, 3).map((criteria, criteriaIndex) => (
+                        <EuiFlexItem key={criteria.type + criteriaIndex} grow={false}>
+                          <QueryRuleDraggableItemCriteriaDisplay criteria={criteria} />
```
Unrelated to this change, but the query rules list component doesn't announce list items properly on voice over. It hops between "Drag Handle" announcements. Perhaps there's already an a11y issue for this

wildemat reviewed (approved):
approved with some non blocking comments

Samiul-TheSoccerFan reviewed (approved):
LGTM, @wildemat has some good suggestions if you are interested :).

elasticmachine committed:
Merge branch 'main' into searchable-detail-panel

efegurkan committed:
Review changes

Add context to the aria label on draggable element.
Add a max visible constant for the slice of criterion.
Updated filter function to a more readable suggestion from Matt.

efegurkan committed:
Improve aria label for search results

wildemat reviewed (approved):
+1 to @TattdCodeMonkey comments. Otherwise recent changes LGTM

efegurkan committed:
Review changes

kibanamachine commented:
Starting backport for target branches: 8.19, 9.1

https://github.com/elastic/kibana/actions/runs/17774122582

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|Backport failed because of merge conflicts|
|❌|9.1|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 232579
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

efegurkan commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.1|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/235348">](https://github.com/elastic/kibana/pull/235348)|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/235349">](https://github.com/elastic/kibana/pull/235349)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.2 -->
